### PR TITLE
Shadow convert

### PR DESCRIPTION
**Pull request by austinpaine**
Basically just ignore this whole file change, made a bit of a giant mess trying to get the commit history from master into this branch because i did a new git repo since i created a fresh luminus project with +shadow-cljs.  This now has the exact same functionality as sample4 did, but now it runs with shadow-cljs instead of figwheel for client side.

### CODE REVIEW

**Comment by austinpaine** [comment]
If you want to just check the readme and make sure it directs you how to run it right, thats the main piece.

**Review by cawasser — COMMENTED** [review]
this also includes all the hidden .idea files too, which we don't want. Not sure it its best, but we might want to redo this once again to clear that up as well. There are like 200 of them...

File: README.md [hunk 1]
```diff
@@ -12,10 +11,19 @@ You will need [Leiningen][1] 2.0 or above installed.
 
 ## Running
 
-To start a web server for the application, run:
+To start the web server for the application, in a terminal/powershell run:
 
     lein run 
 
+To start the client server, in a terminal powershell run `npm install`.
```
Comment (cawasser) [inline README.md:18]: this looks correct, but it's hard to find among all the .idea files

**Comment by austinpaine** [comment]
Alright, fixed up gitignore, and removed the idea and vscode folders with extraneous files.  Looking much better now.

File: README.md (hunk 1 above)
Comment (cawasser) [reply README.md:18]: thanks for cleaning this up

File: README.md [hunk 2]
```diff
@@ -12,10 +11,19 @@ You will need [Leiningen][1] 2.0 or above installed.
...
+    shadow-cljs watch app
+
+This will spin up the clientside server hosting the webpage.
+You should now be able to open a browser window to `localhost:3000`
+and see the luminus homepage with various welcome banners.
+
 ## License
 
-Copyright © 2019 FIXME
+Copyright © 2020 Austin Paine
```
Comment (cawasser) [inline README.md]: lets drop either go back to the FIXME, or drop it completely. Doesn't seem correct to use any individual on this.

File: shadow-cljs.edn [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
+ ;;
+ ;; >>>>>>>>>>>>>>>>>>>>>>>>>> DO NOT EDIT THIS FILE <<<<<<<<<<<<<<<<<<<<<<<<<<<
```
Comment (cawasser) [inline shadow-cljs.edn]: we can probably drop this warning block as long as we mention in the readme to only user lean-shadow ONCE, and then NEVER AGAIN!

File: README.md [hunk 4]
```diff
@@ -12,10 +11,19 @@ You will need [Leiningen][1] 2.0 or above installed.
 
 ## Running
 
-To start a web server for the application, run:
+To start the web server for the application, in a terminal/powershell run:
 
     lein run 
 
+To start the client server, in a terminal powershell run `npm install`.
+When that finishes, run:
```
Comment (cawasser) [inline README.md:19]: do we need to run lean-shadow once before the first dev build, or has the template already built the shadow-cljs.edn file already?

File: env/dev/clj/user.clj [hunk 5]
```diff
@@ -1,21 +1,31 @@
...
     [mount.core :as mount]
-    [sample-4.figwheel :refer [start-fw stop-fw cljs]]
-    [sample-4.core :refer [start-app]]))
+    [sample4.core :refer [start-app]]))
 
 (alter-var-root #'s/*explain-out* (constantly expound/printer))
 
-(defn start []
-  (mount/start-without #'sample-4.core/repl-server))
+(add-tap (bound-fn* clojure.pprint/pprint))
```
Comment (cawasser) [inline env/dev/clj/user.clj:13]: we haven't used tap> before. probably need to research this a bit and add to the instructions

**Review by cawasser — APPROVED** [review]

**Pull request merged**